npm/webpack-preprocessor/index.ts

```diff
@@ -7,6 +7,39 @@ import webpack from 'webpack'
 import utils from './lib/utils'
 import { overrideSourceMaps } from './lib/typescript-overrides'
 
+const getTsLoaderIfExists = (rules) => {
+  let tsLoaderRule
+
+  rules.some((rule) => {
+    if (!rule.use && !rule.loader) return false
+
+    if (Array.isArray(rule.use)) {
+      const foundRule = rule.use.find((use) => {
+        return use.loader && use.loader.includes('ts-loader')
+      })
+
+      tsLoaderRule = foundRule
+
+      return tsLoaderRule
+    }
+
+    if (_.isObject(rule.use) && rule.use.loader && rule.use.loader.includes('ts-loader')) {
+      // it's an Object
```

Comment:
Not sure this comment is really that helpful...

Reply:
i need to add some better comments in here as to what we are checking for